client/pages/_app.js

```diff
@@ -1,16 +1,19 @@
 import { DefaultSeo } from 'next-seo';
 import SEO from '@/next-seo.config';
-import { Wrapper } from '@/components';
+import { Wrapper, Modal } from '@/components';
 import '@/styles/main.scss';
-import { ContextProvider } from '@/utils/context';
+import { ModalProvider } from '@/utils/modalContext'; 
```

Comment:
IF we only have the modal state in this context, we can just add it to the context provider, no need to have a separate context just for the Modal